packages/components/rollup.config.mjs

```diff
@@ -92,40 +120,45 @@ export default {
           src: "src/Combobox/components/ComboboxContent/ComboboxContentList/ComboboxContent.css.d.ts",
           dest: "dist/Combobox/components/ComboboxContent/ComboboxContentList",
         },
+        {
+          src: "dist/index.d.ts",
+          dest: "dist",
+          rename: "index.d.mts",
+        },
       ],
+      hook: "writeBundle",
     }),
   ],
   output: [
     {
       dir: "dist",
+      entryFileNames: "[name].cjs",
+      exports: "named",
       format: "cjs",
     },
+    {
+      dir: "dist",
+      entryFileNames: "[name].mjs",
+      format: "esm",
+    },
   ],
   external: [
     "react",
     "react-hook-form",
     "react-router-dom",
     "react-dom",
-    "react-markdown",
-    "react-countdown",
     "react-popper",
-    "react-datepicker",
-    "react-dropzone",
     "react-dom/client",
     "axios",
     "filesize",
     "color",
     "framer-motion",
     "classnames",
-    new RegExp("lodash/.*"),
     "@std-proposal/temporal",
     "@jobber/design",
     "@jobber/design/foundation",
     "@jobber/formatters",
-    new RegExp("@jobber/hooks/.*"),
     "zxcvbn",
-    "use-resize-observer/polyfilled",
-    "@apollo/client",
```

Comment:
Why were some of these removed and others not?

Reply:
Good question!

These packages caused issues with jobber-frontend when we packaged them this way after getting the ESM piece working there.

I think we're slightly misusing the 'external' configuration option for Rollup, because if we're making something as external, I believe we also have to mark it as a peerDep inside the package.json.

I think we're getting away with it because some of the packages we've marked as external (so rollup doesn't bundle them) just happen to be present in node_modules because someone else manually included that package (say inside JO), or because they were included by a 3rd party library.

I wanted to go the full mile and either mark everything in here as a peerDep inside package.json, or remove them from here entirely but I didn't want to overload this PR with unrelated changes.

I definitely think we should circle back here when we get some tech debt time, to clean up out we're bundling/exporting/peerdepping 3rd party libraries. I think if we moved `react-hook-forms` in here specifically into a peerDep inside package.json, we could fix the issues people were having with trying to roll their own react-hook-forms context.